src/app/interaction-model/IterationTypes.h

```diff
@@ -35,14 +35,21 @@ enum class ClusterQualityFlags : uint32_t
 
 struct ClusterInfo
 {
-    DataVersion dataVersion; // current version of this cluster
+    DataVersion dataVersion = 0; // current cluster data version
```

Comment:
0 is not a special value for dataversion; it's a perfectly valid value.  So if we have any situations in which this can be used without the dataVersion being initialized, setting it to 0 here is just going to cover up bugs, no?

Why this change?

Reply:
My thinking was that this is at least predictable. While any value is valid, getting "whatever is on the stack" seems more likely to be odd compared to a fixed value. Was going with "constructors should always initialise data" and 0 seems natural.

The other alternative seems to make the default constructor private and enforce a version on construction, however that seems more annoying to work with.

Reply:
Should it be the other way around where it is initialized to the max type value as chances are we will never get to this dataVersion

Reply:
I can also make it some uint max, however since dataversion is supposed to wrap, I am unsure if worth it. 
Do we gain benefits from large data versions as a starting point?

Reply:
@jmartinez-silabs All values are equally likely; data versions are set to a random 32-bit integer on startup.

@andy31415 If it's feasible to not produce invalid data, we should try to do that.  Either make DataVersion an optional (to make it clear that it might not be set/known for some reason) or yes, make the constructor require a DataVersion value.

Reply:
Updated to make data version mandatory and to propagate/log errors.